### PR TITLE
chore: export external types

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR exports external types so that consumers do not have to import modules like unixfs to access them

Package lock was also updated

### CODE REVIEW

File: src/pack/blob.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 import all from 'it-all'
-import { ImportCandidateStream } from 'ipfs-core-types/src/utils'
+import { ImportCandidateStream as ImportCandidateStreamT } from 'ipfs-core-types/src/utils'
 
 import { Blockstore } from '../blockstore'
 import { MemoryBlockStore } from '../blockstore/memory'
 
 import { pack } from './index'
 
+export type ImportCandidateStream = ImportCandidateStreamT
```
Comment (olizilla) [inline src/pack/blob.ts:10]: I think we could do something like:
```suggestion
export type { ImportCandidateStream }
```

see: https://github.com/ipfs-shipyard/nft.storage/blob/bc0201cb7f8d2b4a6114f3071203c4e685a14066/packages/client/src/lib/interface.ts#L1-L2

**Review by olizilla — CHANGES_REQUESTED** [review]
I think we don't need to rename our imports if we use an export statement like
```
import { Thing }
export { Thing }
```
https://www.typescriptlang.org/docs/handbook/modules.html#export-statements

File: src/pack/blob.ts (hunk 1 above)
Comment (vasco-santos) [reply src/pack/blob.ts]: oh I see, that is because of the `import type`. Without it TS complains

Good catch

File: src/unpack/index.ts [hunk 2]
```diff
@@ -5,10 +5,11 @@ import { CarIndexedReader, CarReader } from '@ipld/car'
 import { Block } from '@ipld/car/api'
 import { CID } from 'multiformats'
 import exporter from '@vascosantos/ipfs-unixfs-exporter'
-import { UnixFSEntry } from '@vascosantos/ipfs-unixfs-exporter'
+import type { UnixFSEntry } from '@vascosantos/ipfs-unixfs-exporter'
+export type { UnixFSEntry }
 
 // Export unixfs entries from car file
-export async function* unpack (carReader: CarReader|CarIndexedReader, roots?: CID[]): AsyncIterable<UnixFSEntry> {
+export async function* unpack(carReader: CarReader | CarIndexedReader, roots?: CID[]): AsyncIterable<UnixFSEntry> {
```
Comment (olizilla) [inline src/unpack/index.ts:12]: We could simplify this signature too now that the CarReader type is an interface in https://github.com/ipld/js-car/pull/32/files

**Review by olizilla — APPROVED** [review]
LGTM! I need this!

**Pull request merged**